src/Init/Lean/PrettyPrinter/Parenthesizer.lean

```diff
@@ -0,0 +1,444 @@
+/-
+Copyright (c) 2020 Sebastian Ullrich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+-/
+
+/-!
+The parenthesizer inserts parentheses into a `Syntax` object where syntactically necessary, usually as an intermediary
+step between the delaborator and the formatter. While the delaborator outputs structurally well-formed syntax trees that
```

Comment:
I was assuming the delaborator was **not** going to produce structurally well-formed syntax due to missing parentheses.  So, is the plan to add aggressively add parentheses in the delaborator?

Reply:
No, there is no change of plans. The *tree structure* of the delaborator output is fine, it's only when linearizing this tree into e.g. a single string in the formatter that missing parentheses get problematic. Maybe a different word would be a better fit here?